### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by Konstantin8105** [elliotchance/c2go#332]
Now, In according to next code:
https://github.com/elliotchance/c2go/blob/cc45e99c433c5e3c942d84f3015e3b2f385bda26/transpiler/transpiler.go#L321-L353

For case `case *ast.FunctionDecl: ` error is not ignored. So, if somethink wrong inside function(for example, casting, or ...), then function is not included and silent.
Few time ago, I create `typedef struct` and at the on of implementation step - function `main()` is gone from output Go code.

Solution:
* Add warning with description of error
* Add error handling for other cases

**Comment by elliotchance** [comment]
yes. This definitely needs to be raised as an `// Error`. The script still should proceed because outputting something is better than outputting nothing.

**Comment by Konstantin8105** [comment]
So, if we have `err != nil` , then we add Error (it is like Warning now) and add err = nil for avoid empty output,
:+1: Nice.

**Comment by Konstantin8105** [comment]
For all warnings, errors, panics need print `node.Position` for simplification of search.
Because, now - it is not clear:
```
panic: invalid identity: 'U0 *SliceToU0Slice' [recovered]
	panic: bad type: 'noarch.U0 *SliceToU0Slice'

goroutine 1 [running]:
github.com/elliotchance/c2go/util.typeToExpr.func1(0xc421233200, 0x19)
	/home/lepricon/go/src/github.com/elliotchance/c2go/util/goast.go:41 +0x115
panic(0x5a1480, 0xc421234da0)
```
